data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateGroupManager.java

```diff
@@ -5,27 +5,53 @@
 
 package com.amazon.dataprepper.plugins.processor.aggregate;
 
-import java.util.Map;
-
 import com.google.common.collect.Maps;
 
-class AggregateGroupManager {
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
-    private final Map<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> allGroupStates = Maps.newConcurrentMap();
+class AggregateGroupManager {
 
-    AggregateGroupManager() {
+    private final Map<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> allGroups = Maps.newConcurrentMap();
+    private final int windowDuration;
 
+    AggregateGroupManager(final int windowDuration) {
+        this.windowDuration = windowDuration;
     }
 
     AggregateGroup getAggregateGroup(final AggregateIdentificationKeysHasher.IdentificationHash identificationHash) {
-        final AggregateGroup aggregateGroup = allGroupStates.get(identificationHash);
+        final AggregateGroup aggregateGroup = allGroups.get(identificationHash);
 
         if (aggregateGroup != null) {
             return aggregateGroup;
         }
 
         final AggregateGroup newAggregateGroup = new AggregateGroup();
-        allGroupStates.put(identificationHash, newAggregateGroup);
+        allGroups.put(identificationHash, newAggregateGroup);
         return newAggregateGroup;
     }
+
+    List<Map.Entry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup>> getGroupsToConclude() {
+        final List<Map.Entry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup>> groupsToConclude = new ArrayList<>();
+        for (final Map.Entry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> groupEntry : allGroups.entrySet()) {
+            if (shouldConcludeGroup(groupEntry.getValue())) {
+                groupsToConclude.add(groupEntry);
+            }
+        }
+        return groupsToConclude;
+    }
+
+    private boolean shouldConcludeGroup(final AggregateGroup aggregateGroup) {
+        return (Instant.now().toEpochMilli() - aggregateGroup.getGroupStart().toEpochMilli()) >= windowDuration * 1000L;
```

Comment:
You could convert the `windowDuration` to either Java 8 `Duration` in the constructor. Then you could perform direct date-time comparisons on both `Instants`. It would be a little cleaner. But, it's an optional change.